pkg/apis/operator/v1alpha1/tektonresult_validation.go

```diff
@@ -33,6 +38,20 @@ func (tp *TektonResult) Validate(ctx context.Context) (errs *apis.FieldError) {
 		errMsg := fmt.Sprintf("metadata.name, Only one instance of TektonResult is allowed by name, %s", ResultResourceName)
 		return errs.Also(apis.ErrInvalidValue(tp.GetName(), errMsg))
 	}
+	errs = errs.Also(tp.Spec.validate("spec"))
+	return errs
+}
 
-	return nil
+func (trs *TektonResultSpec) validate(path string) (errs *apis.FieldError) {
+	if trs.LokiStackName != "" {
+		if strings.ToLower(trs.LogsType) != LogsTypeLoki {
+			errMsg := fmt.Sprintf("Loki stack is only supported when logs_type is loki or empty, got logs_type: %s", trs.LogsType)
+			errs = errs.Also(apis.ErrInvalidValue(trs.LogsType, fmt.Sprintf("%s.logs_type", path), errMsg))
+		}
+		if trs.LokiStackNamespace == "" {
+			errMsg := "Loki stack namespace is required when loki_stack_name is provided"
+			errs = errs.Also(apis.ErrInvalidValue(trs.LokiStackNamespace, fmt.Sprintf("%s.loki_stack_namespace", path), errMsg))
+		}
+	}
+	return errs
```

Comment:
@khrm Why do you verify `LokiStackName`, if it is found then verifying the `type`?

`type` is the root for log config. your verification should be based on the `type`.

Reply:
Check the documentation. In the case of LokiStack, the user doesn't need to provide results API config.

Now, if he has provided a type, it should be `loki` only.